src/ET_Client.php

```diff
@@ -338,17 +407,7 @@ function refreshTokenWithOAuth2($forceRefresh = false)
 
                 $url = $this->baseAuthUrl."/v2/token";
 
-                $jsonRequest = new stdClass();
-                $jsonRequest->client_id = $this->clientId;
-                $jsonRequest->client_secret = $this->clientSecret;
-                $jsonRequest->grant_type = "client_credentials";
-
-                if ($this->isNullOrEmptyString($this->accountId) == false){
-                    $jsonRequest->account_id = $this->accountId;
-				}
-				if ($this->isNullOrEmptyString($this->scope) == false){
-                    $jsonRequest->scope = $this->scope;
-				}
+                $jsonRequest = $this->createPayloadForOauth2();
```

Comment:
This variable name should be renamed to $jsonPayload or just $payload since this is what it represents and not a request.